server/master/src/main/java/org/apache/accumulo/master/recovery/RecoveryManager.java

```diff
@@ -51,19 +53,33 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.Weigher;
+
 public class RecoveryManager {
 
   private static final Logger log = LoggerFactory.getLogger(RecoveryManager.class);
 
   private Map<String,Long> recoveryDelay = new HashMap<>();
   private Set<String> closeTasksQueued = new HashSet<>();
   private Set<String> sortsQueued = new HashSet<>();
+  private Cache<Path,Boolean> existenceCache;
   private ScheduledExecutorService executor;
   private Master master;
   private ZooCache zooCache;
 
-  public RecoveryManager(Master master) {
+  public RecoveryManager(Master master, long timeToCacheExistsInMillis) {
     this.master = master;
+    existenceCache =
+        CacheBuilder.newBuilder().expireAfterWrite(timeToCacheExistsInMillis, TimeUnit.MILLISECONDS)
+            .maximumWeight(10000000).weigher(new Weigher<Path,Boolean>() {
+              @Override
+              public int weigh(Path path, Boolean exist) {
+                return path.toString().length();
+              }
+            }).build();
```

Comment:
Weigher is a functional interface. You can use a lambda here. Also, you can use underscores to make the number more readable:

```java
    .maximumWeight(10_000_000).weigher((path, exist) -> path.toString().length()).build();
```

Reply:
This is 1.9, so no lambda.

Reply:
The suggestion to use underscores is still relevant for 1.9.

Reply:
Used underscores in 7ad5256